R/run_ml.R

```diff
@@ -201,30 +204,43 @@ run_ml <-
     }
     check_training_frac(training_frac)
     check_training_indices(training_inds, dataset)
-
+    
     train_data <- dataset[training_inds, ]
     test_data <- dataset[-training_inds, ]
+    if (impute_in_training == TRUE) {
+      
+      train_processed_data <- prep_data(train_data, outcome_colname, prefilter_threshold=1, method=c("center", "scale"), impute_in_preprocessing=TRUE, to_numeric=TRUE)
+      train_processed_feats <- train_processed_data$processed_feats
+      split_dat <- train_processed_data$split_dat
+      train_data <- dplyr::bind_cols(split_dat$outcome, train_processed_feats) %>%
+        dplyr::as_tibble()
+      test_processed_data <- prep_data(test_data, outcome_colname, prefilter_threshold=1, method=c("center", "scale"), impute_in_preprocessing=TRUE, to_numeric=TRUE)
+      test_processed_feats <- test_processed_data$processed_feats
+      split_dat <- test_processed_data$split_dat
+      test_data <- dplyr::bind_cols(split_dat$outcome, test_processed_feats) %>%
+        dplyr::as_tibble()
+    }
```

Comment:
As you now have it, `prep_data()` does more than just impute missing values -- it also does other preprocessing steps as advertised by `preprocess_data()`. 

So it seems like there are two potential implementations of this feature:

A) Users can opt to do all preprocessing, including imputation, after the train/test split.
B) Users can opt to do only imputation and not other preprocessing after the train/test split. In this instance, presumably they perform all preprocessing minus imputation before calling `run_ml`.
C) As it is now, users could perform the train/test split on their own, do preprocessing on the splits separately (which includes imputation), and assign the train/test sets in `run_ml()` explicitly by providing a vector of indices to `train_frac`.

This opens a wider discussion about how we want the API to work, and how much flexibility we want to give users. 

What this code currently provides is an easy way to do option A. What I initially had in mind for this PR is option B, because option A could be done via option C.

Reply:
We reworked the code so it does what you described in option B now.